workflow/gccontroller/gc_controller.go

```diff
@@ -24,21 +26,31 @@ import (
 )
 
 type Controller struct {
-	wfclientset wfclientset.Interface
-	wfInformer  cache.SharedIndexInformer
-	workqueue   workqueue.DelayingInterface
-	clock       clock.Clock
-	metrics     *metrics.Metrics
+	wfclientset           wfclientset.Interface
+	wfInformer            cache.SharedIndexInformer
+	workqueue             workqueue.DelayingInterface
+	clock                 clock.Clock
+	metrics               *metrics.Metrics
+	completedOrderedQueue *gcHeap
+	failedOrderedQueue    *gcHeap
+	errorOrderedQueue     *gcHeap
+	retentionPolicy       *config.RetentionPolicy
+	limiter               <-chan time.Time
```

Comment:
not clear what this is for from name alone

Reply:
Update and move it to a variable.